libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/NavigationMapRoute.java

```diff
@@ -162,272 +214,528 @@ public NavigationMapRoute(@Nullable MapboxNavigation navigation, @NonNull MapVie
    */
   private void initialize() {
     layerIds = new ArrayList<>();
-
-    addSource(route);
-
-    TypedArray typedArray = mapView.getContext().obtainStyledAttributes(styleRes, R.styleable.NavigationMapRoute);
-
-    routeDefaultColor = typedArray.getColor(R.styleable.NavigationMapRoute_routeColor,
-      ContextCompat.getColor(mapView.getContext(), R.color.mapbox_navigation_route_layer_blue));
-    routeModerateColor = typedArray.getColor(R.styleable.NavigationMapRoute_routeModerateCongestionColor,
-      ContextCompat.getColor(mapView.getContext(), R.color.mapbox_navigation_route_layer_congestion_yellow));
-    routeSevereColor = typedArray.getColor(R.styleable.NavigationMapRoute_routeSevereCongestionColor,
-      ContextCompat.getColor(mapView.getContext(), R.color.mapbox_navigation_route_layer_congestion_red));
-    @ColorInt int routeShieldColor = typedArray.getColor(R.styleable.NavigationMapRoute_routeShieldColor,
-      ContextCompat.getColor(mapView.getContext(), R.color.mapbox_navigation_route_shield_layer_color));
-    float routeScale = typedArray.getFloat(R.styleable.NavigationMapRoute_routeScale, 1.0f);
-
-    addNavigationRouteLayer(routeScale);
-    addNavigationRouteShieldLayer(routeShieldColor, routeScale);
-    typedArray.recycle();
-  }
-
-  /**
-   * Adds the necessary listeners
-   */
-  private void addListeners() {
-    if (navigation != null) {
-      navigation.addProgressChangeListener(this);
-    }
-    mapView.addOnMapChangedListener(this);
+    getAttributes();
+    placeRouteBelow();
   }
 
   /**
-   * Pass in a {@link DirectionsRoute} and display the route geometry on your map.
+   * Allows adding a single primary route for the user to traverse along. No alternative routes will
+   * be drawn on top of the map.
    *
-   * @param route a {@link DirectionsRoute} used to draw the route line
+   * @param directionsRoute the directions route which you'd like to display on the map
    * @since 0.4.0
    */
-  public void addRoute(@NonNull DirectionsRoute route) {
-    this.route = route;
-    addSource(route);
-    setLayerVisibility(true);
+  public void addRoute(DirectionsRoute directionsRoute) {
+    List<DirectionsRoute> routes = new ArrayList<>();
+    routes.add(directionsRoute);
+    addRoutes(routes);
   }
 
   /**
-   * Remove the route line from the map style, note that this only changes the visibility and does not remove any layers
-   * or sources.
+   * Provide a list of {@link DirectionsRoute}s, the primary route will default to the first route
+   * in the directions route list. All other routes in the list will be drawn on the map using the
+   * alternative route style.
    *
-   * @since 0.4.0
+   * @param directionsRoutes a list of direction routes, first one being the primary and the rest of
+   *                         the routes are considered alternatives.
+   * @since 0.8.0
```

Comment:
Should these `@since` tags be `0.7.1`?

Reply:
I have been tagging everything at 0.8.0 and can go back right before the release and change them. Technically, this PR should be going in the 0.8 release since 0.7.1 is just meant for bug fixes (no additional features). Not sure what we want to do here, but we can always come back and change these later down the road.

Reply:
Okay yeah that's fine with me, just wanted to double check.